src/ConflictResolver/AutoIncrementConflictResolver.php

```diff
@@ -40,20 +42,28 @@ public function __construct(AdapterInterface $adapter)
      */
     public function resolveConflict(UriContext $uriContext)
     {
-        $this->inc = 0;
+        $this->index = 0;
+        $candidateUri = $uri = $uriContext->getUri();
 
-        $uri = $uriContext->getUri();
-        $candidateUri = $this->incrementUri($uri);
-
-        while ($route = $this->adapter->findRouteForUri($candidateUri, $uriContext)) {
+        while ($this->isUriConflicting($candidateUri, $uriContext)) {
             $candidateUri = $this->incrementUri($uri);
         }
 
         return $candidateUri;
     }
 
+    /**
+     * Tell if the given URI for the given context is conflicting with another
+     * route.
+     */
+    protected function isUriConflicting($uri, UriContext $uriContext)
+    {
+        return null !== $uriContext->getCollection()->getAutoRouteByUri($uri)
+            || null !== $this->adapter->findRouteForUri($uri, $uriContext);
```

Comment:
Maybe we can pass the `UriGenerator` as an additional argument and add a `isUriConflicting` method to it ? As it already has a `resolveConflict` method, I think it will not break the SRP.